source/concepts/applications.rst

```diff
@@ -0,0 +1,569 @@
+server can implement to give the clients some control over how the frames are
+streamed. Finally, it defines some interactions with the multiplayer
+application to share where to display the molecular system relative to the
+users, and how to render the molecules.
+
+Frames
+~~~~~~
+
+The :ref:`trajectory service <trajectory-service>` allows to stream snapshots
+of arbitrary data to clients. Each snapshot is described in a :ref:`FrameData
+<frame-description>` which contains a key-value map of protobuf `Values
+<https://protobuf.dev/reference/protobuf/google.protobuf/#value>`_ and one of
+homogeneous arrays. Here, we define a set of keys and data format to describe
+the semantics of molecular systems.
+
+.. note::
+
+   A server using this set of keys can implement keys from another application
+   as well. For instance, a server implementing the :ref:`iMD application
+   <imd-application>` can implement both this set of keys and :ref:`iMD-specific
+   keys <imd-framedata-keys>`.
+
+All FrameData values used by the trajectory application use the following set
+of units:
+
+* lengths are expressed in nanometers (:math:`\text{nm}`)
+* durations are expressed in picoseconds (:math:`\text{ps}`)
+* masses are expressed in atomic mass unit (AMU)
+* charges are expressed in proton charge
+* energies are expressed in :math:`\text{kJ}\cdot\text{mol}^{-1}`
+* velocities are expressed in :math:`\text{nm}\cdot{ps}^{-1}`
+* forces are expressed in :math:`\text{kJ}\cdot\text{mol}^{-1}\cdot\text{nm}^{-1}`
+
+The coordinate system is the right-handed, Z-up, system used in most software
+working with molecular systems.
+
+Particles
+^^^^^^^^^
+
+A molecular system is composed of atoms. The application refers to them as
+"particles" to account for representations that do not deal with individual
+atoms, such as coarse-grained models (`e.g.` `Martini <http://cgmartini.nl/>`_
+or `SIRAH <http://www.sirahff.com/>`_). Particles are described by the following
+keys in the array map:
+
+* ``particle.positions``: the Cartesian coordinates of each particle. The
+  coordinates are stored as a flat array of coordinates where each triplet
+  corresponds to the XYZ coordinates of a particle.
+* ``particle.velocities``: the velocity of each particle. Like the positions,
+  they are expressed as a flattened array of triplets.
+* ``particle.forces``: the force array applied to each particle, as a flattened
+  array of triplets.
+* ``particle.elements``: the chemical element for each particle expressed as
+  atomic numbers. If a particle is not an atom, or if a chemical element is not
+  relevant for any reason, the atomic number can be set to 0.
+* ``particle.names``: a name for each particle. Each name is an arbitrary string
+  to identify the particle, usually within a residue. If an atom does not have
+  a name, set it to an empty string. When applicable, it is recommended to use
+  the names used in the Protein Data Bank.
+
+.. warning::
```

Comment:
@hjstroud Does this warning cover your question from earlier about leap frog and velocities?

Reply:
I think this is good! Maybe it's worth making clear that calculating the velocities are half a _time_ step out of sync? I've suggested a change below, what do you think?